File: backend/models/notificationModel.js
const mongoose = require('mongoose');

const notificationSchema = new mongoose.Schema({
  type: {
    type: String,
    required: [true, 'A notification must have a type'],
    enum: ['lesson', 'quiz', 'community', 'system', 'announcement', 'support'],
  },
  message: {
    type: String,
    required: [true, 'A notification must have a message'],
  },
  link: {
    type: String,
  },
  userId: {
    type: mongoose.Schema.ObjectId,
    ref: 'User',
    required: [true, 'A notification must belong to a user'],
  },
  adminId: {
    type: mongoose.Schema.ObjectId,
    ref: 'User',
  },
  teacherId: {
    type: mongoose.Schema.ObjectId,
    ref: 'User',
  },
  read: {
    type: Boolean,
    default: false,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

notificationSchema.index({ userId: 1, read: 1 });
notificationSchema.index({ createdAt: -1 });

const Notification = mongoose.model('Notification', notificationSchema);

module.exports = Notification;

File: mobile/lib/models/notification_model.dart
class NotificationModel {
  final String id;
  final String type; // 'lesson' | 'quiz' | 'community' | 'system' | 'announcement'
  final String message;
  final String? link;
  final bool read;
  final DateTime createdAt;

  const NotificationModel({
    required this.id,
    required this.type,
    required this.message,
    this.link,
    required this.read,
    required this.createdAt,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['_id'] ?? json['id'] ?? '',
      type: json['type'] ?? 'system',
      message: json['message'] ?? '',
      link: json['link'],
      read: json['read'] as bool? ?? false,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString()) ?? DateTime.now()
          : DateTime.now(),
    );
  }
}
